source/application/BioXAS/BioXASAppController.cpp

```diff
@@ -635,6 +631,16 @@ AMScanConfigurationViewHolder3* BioXASAppController::createScanConfigurationView
 	return configurationViewHolder;
 }
 
+void BioXASAppController::addComponentView(QObject *component, const QString &viewName, const QString &categoryName, const QString &icon)
```

Comment:
Is it possible to move this function to the `CLSAppController`? Then we can have the `componentViewMapping_` in CLSAppController, add add a function to do the following mapping:
```
    QWidget *detectorView = componentViewMapping_.value(detector, 0);
    QWidget *detectorPane = viewPaneMapping_.value(detectorView, 0);
```

Maybe the other beamlines can use it as well.